### PR TITLE
fix(core): ensure grid keyboard navigation works

### PR DESCRIPTION
**Pull request by khotcholava**
fix(core): ensure grid keyboard navigation works

closes [#12198](https://github.com/SAP/fundamental-ngx/issues/12198) 

## Description
Implemented f2 tab arrow and f7 navigations

### CODE REVIEW

File: libs/core/grid-list/components/grid-list-item/grid-list-item.component.ts [hunk 1]
```diff
@@ -144,6 +153,7 @@ export class GridListItemComponent<T> implements AfterViewInit, OnDestroy {
     set type(value: Nullable<GridListItemType>) {
         this._type = value ?? 'inactive';
     }
+
```
Comment (dpavlenishvili) [inline libs/core/grid-list/components/grid-list-item/grid-list-item.component.ts:156]: add notes top of the set and get

File: libs/cdk/utils/directives/focusable-item/focusable-item.directive.ts [hunk 2]
```diff
@@ -120,6 +120,7 @@ export class FocusableItemDirective implements FocusableItem, HasElementRef {
             fromEvent(this.elementRef.nativeElement, 'focusout')
                 .pipe(takeUntilDestroyed())
                 .subscribe(() => {
+                    console.log('focyseout');
```
Comment (dpavlenishvili) [inline libs/cdk/utils/directives/focusable-item/focusable-item.directive.ts:123]: remove log

File: libs/core/grid-list/components/grid-list/grid-list.component.ts [hunk 3]
```diff
@@ -121,7 +143,8 @@ export class GridListComponent<T> extends GridList<T> implements OnChanges, Afte
     /** @hidden */
     constructor(
         private readonly _cd: ChangeDetectorRef,
-        private _elRef: ElementRef
+        private _elRef: ElementRef,
```
Comment (dpavlenishvili) [inline libs/core/grid-list/components/grid-list/grid-list.component.ts:146]: just to maintain consistency add readonly for injections

File: libs/core/grid-list/components/grid-list/grid-list.component.ts [hunk 4]
```diff
@@ -231,16 +261,10 @@ export class GridListComponent<T> extends GridList<T> implements OnChanges, Afte
...
-            } else if (
-                event.shiftKey &&
-                this.selectionMode === 'multiSelect' &&
-                KeyUtil.isKeyCode(event, [HOME, END])
-            ) {
-                this._handleHomeEndKeydown(event);
-            }
+        if (KeyUtil.isKeyCode(event, [UP_ARROW, DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW])) {
+            this._handleArrowKeydown(event);
+        } else if (event.shiftKey && this.selectionMode === 'multiSelect' && KeyUtil.isKeyCode(event, [HOME, END])) {
```
Comment (dpavlenishvili) [inline libs/core/grid-list/components/grid-list/grid-list.component.ts:266]: add enum for selectionMode

File: libs/core/grid-list/models/grid-list-focus-item.model.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import { Subject } from 'rxjs';
+
+@Directive()
+export abstract class GridListFocusItem implements KeyboardSupportItemInterface {
+    /** tab index attribute */
+    @Input()
+    set tabIndex(value: number) {
+        this._tabIndex$.set(coerceNumberProperty(value, -1));
+    }
+
```
Comment (dpavlenishvili) [inline libs/core/grid-list/models/grid-list-focus-item.model.ts:13]: note for getter

File: libs/core/grid-list/models/grid-list-focus-item.model.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    get tabIndex(): number {
```
Comment (dpavlenishvili) [inline libs/core/grid-list/models/grid-list-focus-item.model.ts:14]: note for getter

File: libs/core/list/list.component.ts [hunk 7]
```diff
@@ -180,6 +180,7 @@ export class ListComponent implements ListComponentInterface, ListUnreadIndicato
 
     /** @hidden */
     ngAfterContentInit(): void {
+        console.log(this._focusItems);
```
Comment (dpavlenishvili) [inline libs/core/list/list.component.ts:183]: remove log

File: libs/docs/core/grid-list/examples/combo-select/grid-list-combo-select-example.component.html [hunk 8]
```diff
@@ -11,61 +11,69 @@
...
+                        <span fd-grid-list-title-bar-additional-title-item>{{ '(' + list.length + ')' }}</span>
+                        <fd-grid-list-title-bar-spacer></fd-grid-list-title-bar-spacer>
+                        <button
+                            fd-button
+                            fdType="transparent"
+                            glyph="sort"
+                            aria-labelledby="grid-list-title-bar-1 grid-list-button-1"
+                            aria-label="Sort"
+                            title="Sort"
+                            fdCompact
```
Comment (dpavlenishvili) [inline libs/docs/core/grid-list/examples/combo-select/grid-list-combo-select-example.component.html:26]: remove fdCompact usages

**Comment by khotcholava** [comment]
**NOTE** 
This is the draft PR. 
Still need to fix some small issues

**Comment by khotcholava** [comment]
Closed because new [PR](https://github.com/SAP/fundamental-ngx/pull/12288)

**Pull request closed without merging**